components/reddit/sources/new-comments-by-user/new-comments-by-user.js

```diff
@@ -0,0 +1,95 @@
+const common = require("../../common");
+const { reddit } = common.props;
+
+module.exports = {
+  ...common,
+  key: "new-comments-by-user",
+  name: "New comments by user",
+  description: "Emits an event each time a user posts a new comment.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    reddit,
+    username: { propDefinition: [reddit, "username"] },
+    context: {
+      type: "integer",
+      label: "context",
+      description: "an integer between 2 and 10",
+    },
+    t: { propDefinition: [reddit, "t"] },
+    sr_detail: { propDefinition: [reddit, "sr_detail"] },
+    ...common.props,
+  },
+  hooks: {
+    async deploy() {
+      // Emits sample events on the first run during deploy.
+      let before = null;
+      try {
+        var reddit_comments = await this.reddit.getNewUserComments(
+          null,
+          this.username,
+          this.context,
+          this.t,
+          this.sr_detail,
+          10
+        );
+      } catch (err) {
+        if (common.methods.did4xxErrorOccur(err)) {
+          throw new Error(
+            `We encountered a 4xx error trying to fetch comments by ${this.username}. Please check the username and try again`
```

Comment:
A `4xx` status code can mean a lot of things besides a bad parameter. If the user provided a valid `username` value and we still show this error it will create confusion and a bad customer experience.

I'm inclined to just let it bubble up unless we can do something about it (e.g. do a retry).

Reply:
Ok, bubbling up.